ARCs/arc-draft_extended_desc.md

```diff
@@ -0,0 +1,160 @@
+  keyType: string;
+  /** The type of the value. Can be ABI type or named struct */
+  valueType: string;
+  /** The key itself, as a byte array */
+  key: number[];
+}
+
+interface StorageMap {
+  /** The type of the key. Can be ABI type or named struct */
+  keyType: string;
+  /** The type of the value. Can be ABI type or named struct */
+  valueType: string;
+  /** The prefix of the map, as a string */
+  prefix: string;
+}
+
+interface Method {
+  /** The name of the method */
+  name: string;
+  /** Optional, user-friendly description for the method */
+  desc?: string;
+  /** The arguments of the method, in order */
+  args: Array<{
+    /** The type of the argument */
+    type: string;
+    /** If the type is a struct, the name of the struct */
+    struct?: string;
+    /** Optional, user-friendly name for the argument */
+    name?: string;
+    /** Optional, user-friendly description for the argument */
+    desc?: string;
+  }>;
+  /** Information about the method's return value */
+  returns: {
+    /** The type of the return value, or "void" to indicate no return value. */
+    type: string;
+    /** If the type is a struct, the name of the struct */
+    struct?: string;
+    /** Optional, user-friendly description for the return value */
+    desc?: string;
+  };
+  /** an action is a combination of call/create and an OnComplete */
+  actions: {
+    /** OnCompeltes this method allows when appID === 0 */
+    create: ('NoOp' | 'OptIn' | 'DeleteApplication')[];
+    /** OnCompeltes this method allows when appID !== 0 */
+    call: ('NoOp' | 'OptIn' | 'CloseOut' | 'ClearState' | 'UpdateApplication' | 'DeleteApplication')[];
+  };
+  readonly: boolean;
+}
+
+interface Contract {
+  /** A user-friendly name for the contract */
+  name: string;
+  /** Optional, user-friendly description for the interface */
+  desc?: string;
+  /**
+   * Optional object listing the contract instances across different networks
+   */
+  networks?: {
+    /**
+     * The key is the base64 genesis hash of the network, and the value contains
+     * information about the deployed contract in the network indicated by the
+     * key
+     */
+    [network: string]: {
+      /** The app ID of the deployed contract in this network */
+      appID: number;
+    };
+  };
+  /** Named structs use by the application */
+  structs: StructFields;
+  /** All of the methods that the contract implements */
+  methods: Method[];
+  state: {
+    schema: {
+      global: {
+        ints: number;
+        bytes: number;
+      };
+      local: {
+        ints: number;
+        bytes: number;
+      };
+    };
+    /** Describes single key-value pairs in the application's state */
+    keys: {
+      global: StorageKey[];
+      local: StorageKey[];
+      box: StorageKey[];
+    };
+    /** Describes key-value maps in the application's state */
+    maps: {
+      global: StorageMap[];
+      local: StorageMap[];
+      box: StorageMap[];
+    };
+  };
+}
```

Comment:
Do we need source or bare call config?

Reply:
> source
Yes forgot about that. 

> bare call config?
Yes we probably should, although as a sidenote I think we should encourage developers to avoid using bare calls as much as possible because I've learnt they can be rather confusing to new developers.

Reply:
Fair, but naturally they are still useful for advanced scenarios so we should expose it